nsdb-http/src/main/scala/io/radicalbit/nsdb/web/routes/CommandApi.scala

```diff
@@ -225,55 +278,53 @@ trait CommandApi {
-                          DescribeMetricResponse(
-                            schema.fieldsMap.map {
-                              case (_, field) =>
-                                Field(name = field.name, `type` = field.indexType.getClass.getSimpleName)
-                            }.toSet,
-                            metricInfo
-                          )
+    pathPrefix(Segment) { db =>
+      pathPrefix(Segment) { namespace =>
+        pathPrefix(Segment) { metric =>
+          (pathEnd & get) {
+            withMetricAuthorization(db, namespace, metric, false, authorizationProvider) {
+              onComplete(
+                Future.sequence(
+                  Seq((readCoordinator ? GetSchema(db, namespace, metric)).mapTo[SchemaGot],
+                      (metadataCoordinator ? GetMetricInfo(db, namespace, metric)).mapTo[MetricInfoGot])
+                )) {
+                case Success(SchemaGot(_, _, _, Some(schema)) :: MetricInfoGot(_, _, _, metricInfo) :: Nil) =>
+                  complete(
+                    HttpEntity(
+                      ContentTypes.`application/json`,
+                      write(
+                        DescribeMetricResponse(
+                          schema.fieldsMap.map {
+                            case (_, field) =>
+                              Field(name = field.name, `type` = field.indexType.getClass.getSimpleName)
+                          }.toSet,
+                          metricInfo
                         )
                       )
                     )
-                  case Success(SchemaGot(_, _, _, schemaOpt) :: MetricInfoGot(_, _, _, Some(metricInfo)) :: Nil) =>
-                    complete(
-                      HttpEntity(
-                        ContentTypes.`application/json`,
-                        write(
-                          DescribeMetricResponse(
-                            schemaOpt
-                              .map(s =>
-                                s.fieldsMap.map {
-                                  case (_, field) =>
-                                    Field(name = field.name, `type` = field.indexType.getClass.getSimpleName)
-                                }.toSet)
-                              .getOrElse(Set.empty),
-                            Some(metricInfo)
-                          )
+                  )
+                case Success(SchemaGot(_, _, _, schemaOpt) :: MetricInfoGot(_, _, _, Some(metricInfo)) :: Nil) =>
+                  complete(
+                    HttpEntity(
+                      ContentTypes.`application/json`,
+                      write(
+                        DescribeMetricResponse(
+                          schemaOpt
+                            .map(s =>
+                              s.fieldsMap.map {
+                                case (_, field) =>
+                                  Field(name = field.name, `type` = field.indexType.getClass.getSimpleName)
+                              }.toSet)
```

Comment:
nitpick - here we can use a foldleft to build a set while you create your field? You can avoid to iterate the vector `fieldsMap` twice